packages/playground/src/portal/twin_view.vue

```diff
@@ -25,15 +40,62 @@
             </v-col>
           </v-row>
         </v-list>
+        <v-card-actions v-if="updateRelay" class="justify-end">
+          <v-btn color="primary" class="custom-button" @click="editTwin">Edit</v-btn>
+        </v-card-actions>
       </v-card>
     </v-container>
   </div>
 </template>
 
 <script lang="ts" setup>
+import { generatePublicKey } from "@threefold/rmb_direct_client";
+import { onMounted, ref } from "vue";
+
 import { useProfileManager } from "../stores";
+import { getGrid } from "../utils/grid";
 
 const profileManager = useProfileManager();
+
+const editingTwin = ref(false);
+const relay = ref(profileManager.profile?.relay || "");
+const updateRelay = ref(false);
+const errorMsg = ref("");
+
+onMounted(validateEdit);
+async function validateEdit() {
+  try {
+    const pk = await generatePublicKey(profileManager.profile!.mnemonic);
```

Comment:
Instead of generating a new PK, can we get the existing one from the client? As the value of the generated PK might not match the one on the grid.